knowrob_common/knowrob_common/src/main/java/org/knowrob/utils/MathUtil.java

```diff
@@ -40,6 +40,45 @@
 import javax.vecmath.Vector3d;
 
 public class MathUtil {
+
+    public static double[] transformToMatrix(float[] pose) {
```

Comment:
You can use Matrix4d, it supports to be initialized by quaternion/translation.
Also, there are already predicates `matrix_rotation` and `matrix_translation`,
they could be augmented so that when matrix is bound and rotation/translation is bound
then set the rotation/translation.